packages/propel/src/charts/line-chart/root.tsx

```
@@ -1,107 +1,154 @@
+    legend,
     showTooltip = true,
   } = props;
+  // states
+  const [activeLine, setActiveLine] = useState<string | null>(null);
+  const [activeLegend, setActiveLegend] = useState<string | null>(null);
   // derived values
   const itemKeys = useMemo(() => lines.map((line) => line.key), [lines]);
-  const itemDotClassNames = useMemo(
-    () => lines.reduce((acc, line) => ({ ...acc, [line.key]: line.dotClassName }), {}),
+  const itemLabels: Record<string, string> = useMemo(
+    () => lines.reduce((acc, line) => ({ ...acc, [line.key]: line.label }), {}),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)